core/src/main/java/org/eclipse/hono/config/ProtocolAdapterProperties.java

```diff
@@ -168,15 +169,25 @@ public int getMaxConcurrentConnections() {
     }
 
     /**
-     * Sets the number of maximal concurrent connections the protocol adapter should accept.
+     * Sets the number of maximal concurrent connections the protocol adapter should accept. Setting 0 leaves it to the
+     * protocol adapter to set an appropriate limit.
      * 
-     * @param maxConcurrentConnections The maximal number of concurrent connections.
-     * @throws IllegalArgumentException if the number is &lt;= 0.
+     * @param maxConcurrentConnections The maximal number of concurrent connections, 0 for no explicit limit.
+     * @throws IllegalArgumentException if the number is &lt; 0.
      */
     public void setMaxConcurrentConnections(final int maxConcurrentConnections) {
-        if (maxConcurrentConnections <= 0) {
-            throw new IllegalArgumentException("connection limit must be greater than zero");
+        if (maxConcurrentConnections < 0) {
```

Comment:
the name of this property will be used when configuring the MQTT adapter via environment variables. It would therefore be great if the name could be a little shorter. What about just *maxConnections*? Would that work?
Consequently, can you also please document the configuration variable in the MQTT adapter Admin Guide?